lib/ddtrace/contrib/sidekiq/tracing.rb

```diff
@@ -18,9 +20,24 @@ def initialize(options = {})
         def job_resource(job)
           if job['wrapped']
             job['wrapped']
+          elsif job['class'] == 'Sidekiq::Extensions::DelayedClass'
+            delay_extension_class(job)
           else
             job['class']
           end
+        rescue => e
+          Datadog::Logger.log.debug("Error retrieving Sidekiq job class name (jid:#{job['jid']}): #{e}")
```

Comment:
Side thought, but if this happened on the regular, then user's applications would likely see slowdowns with the raised and handled exceptions, but not have the errors be apparent until they activated debug mode. I don't think there's much to do about it, and overall this is the safest option, though.

Reply:
I'm able to pass the log message as a lazily evaluated block.
I ran some benchmarks and it seems like it's almost 3x as fast when log level is disabled for the message we want to log (the tracer defaults to WARN level, but we are logging a DEBUG message).

I've applied these changes to this PR.

```ruby
Benchmark.ips do |x|
  x.report('block') do
    log.debug { "Error retrieving Sidekiq job class name (jid:#{job['jid']}): #{e}" }
  end
  
  x.report('immediate') do
    log.debug("Error retrieving Sidekiq job class name (jid:#{job['jid']}): #{e}")
  end
end
```

```
Warming up --------------------------------------
               block   307.595k i/100ms
           immediate   127.102k i/100ms
Calculating -------------------------------------
               block      6.441M (±17.4%) i/s -     31.067M in   5.016951s
           immediate      1.780M (± 4.1%) i/s -      8.897M in   5.007571s
```

Reply:
Oh okay fair enough, that wasn't actually what I was pointing to but that works as well. My thought was more about the nature of an exception being repeatedly raised itself (which is expensive), regardless of how its handled. There's no action required on that original thought but your change is fine; thanks!